include/swift/AST/DiagnosticsSema.def

```diff
@@ -3498,14 +3498,21 @@ WARNING(matching_pattern_with_many_assoc_values, none,
         "cannot match several associated values at once, "
         "implicitly tupling the associated values and trying to match that "
         "instead", ())
+NOTE(matching_pattern_with_many_assoc_values_note, none,
+     "'%0' has %1 associated values but there is only 1 pattern here",
```

Comment:
I think we need to rephrase `here` some how, maybe in style of `N associated values cannot be covered by M patterns`?

Reply:
I'm not happy with that note either 😐. However, adding more uses of 'cannot' seems unappealing... the radar author pointed out that writing "cannot" in the warning (just above this one) was already confusing, because (clearly) we _can_ do it (in some cases), it's just that we'd like to avoid doing it.

Reply:
What I'm attempting to suggest here is trying to explain why one pattern is not okay with N associated values instead of just stating that it isn't...

Reply:
Oh! I think I see what you mean. How about

> "The number of patterns should match the number of associated values but '%0' has %1 associated values whereas there is only 1 pattern here"

? Or maybe even split it into two notes

> the number of patterns should match the number of associated values

> '%0' has %1 associated values whereas there is only 1 pattern here

Reply:
I like the version, about about a bit modified version - `number of patterns should match the number of associated values (have %0, expected %1)`?

Reply:
I think mentioning the case is useful for context, whereas the number of patterns in this situation is fixed to be 1 (since we only emit the note when there is a single pattern and two or more associated values). Maybe it could be something like `number of patterns (1) does not match the number of associated values (%1) for the case '%0'`.

Having the numbers at the end like in `(have %0, expected %1)` seems a bit unclear imo -- is it the number of patterns that we have and number of patterns that were expected, or is it the number of associated values that we have and the number of associated values that were expected?

Reply:
Makes sense - if for the N patterns case we can use what you suggested and for a single one we can just say `single pattern does not match ...`, sounds good?